Automattic/newspack-blocks
# feat(donate): additional fields on the form

### All Submissions:






### Changes proposed in this Pull Request:



Adds additional fields feature to the streamlined Donate block:

<img width="810" alt="image" src="https://user-images.githubusercontent.com/7383192/207069863-8e74ddc7-faaa-492c-be16-b5830b9906b8.png">

_(address and phone fields above are added per-block)_

### How to test the changes in this Pull Request:

1. Switch `newspack-plugin` to `feat/donation-additional-fields` branch (https://github.com/Automattic/newspack-plugin/pull/2187)
1. Ensure the site is set up to use Stripe as the Reader Revenue platform and webhooks can reach the site
2. Insert a Donate block on a page. Observe new editor sidebar panel – "Additional data fields"
3. Add some fields
4. Visit the front-end and make a donation, supply values to the additional fields
5. Observe that the fields are added to the Stripe customer metadata – as separate fields, but also the whole group, JSON-encoded, under `_additional_fields` meta key

### Other information:
